js/annotations/navigationBindings.js

```diff
@@ -1098,101 +1102,111 @@ H.setOptions({
+                                        yAxis: 0,
+                                        x: x,
+                                        y: y
+                                    }, {
+                                        xAxis: 0,
+                                        yAxis: 0,
+                                        x: x,
+                                        y: y
+                                    }, {
+                                        xAxis: 0,
+                                        yAxis: 0,
+                                        x: x,
+                                        y: y
+                                    }]
+                                }],
+                                controlPoints: controlPoints
+                            },
+                            navigation
+                                .annotationsOptions,
+                            navigation
+                                .bindings
+                                .rectangleAnnotation
+                                .annotationsOptions
+                        )
+                    );
                 },
                 /** @ignore */
                 steps: [
                     function (e, annotation) {
-                        var xAxis = this.chart.xAxis[0],
-                            yAxis = this.chart.yAxis[0],
-                            inverted = this.chart.inverted,
-                            point = annotation.options.shapes[0].point,
-                            x = xAxis.toPixels(point.x),
-                            y = yAxis.toPixels(point.y),
-                            width = Math.max(
-                                inverted ? e.chartX - y : e.chartX - x,
-                                5
-                            ),
-                            height = Math.max(
-                                inverted ? e.chartY - x : e.chartY - y,
-                                5
-                            );
+                        var points = annotation.options.shapes[0].points,
+                            coords = this.chart.pointer.getCoordinates(e),
+                            x = coords.xAxis[0].value,
+                            y = coords.yAxis[0].value;
+
+                        // Top right point
+                        points[1].x = x;
+                        // Bottom right point (cursor position)
+                        points[2].x = x;
+                        points[2].y = y;
+                        // Bottom left
+                        points[3].y = y;
 
                         annotation.update({
                             shapes: [{
-                                width: width,
-                                height: height,
-                                point: {
-                                    x: point.x,
-                                    y: point.y
-                                }
+                                points: points
```

Comment:
Will the change from multiple properties to the property `points` cause a breaking change in the `Annotation` API?
If so is this something that can cause issues for our users?

Reply:
No, not at all. This change is done only for bindings - not for Annotations API. Bindings: [open demo](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/annotations/gui-buttons/) and add a rectangle - all new created annotations via this button will have different config. Old config is still valid in terms of Annotations API (so loading it from backend will render an annotation anyway).

Reply:
Thanks for the clarification 👍